src/main/java/com/jakub/todoSandbox/repository/JOOQRepository.java

```diff
@@ -0,0 +1,243 @@
+                        .id(record.getId())
+                        .description(record.getDescription())
+                        .priority(Priority.valueOf(record.getPriority().name()))
+                        .steps(fetchSteps(record.getId()))
+                        .build()));
+    }
+
+    @Transactional
+    @Override
+    public Todo saveTodo(Todo todo) throws ValidationException {
+        System.out.println("saveTodo() method call");
+        if (todoService.validateNameAndDesc(todo.name(), todo.description())) {
+            TodoRecord todoRecord = context.insertInto(TODO)
+                    .set(TODO.NAME, todo.name())
+                    .set(TODO.DESCRIPTION, todo.description())
+                    .set(TODO.PRIORITY, PriorityEnum.valueOf(todo.priority().name()))
+                    .returning()
+                    .fetchOne();
+
+            for (Step step : todo.steps()) {
+                assert todoRecord != null;
+                context.insertInto(STEP)
+                        .set(STEP.NAME, step.name())
+                        .set(STEP.DESCRIPTION, step.description())
+                        .set(STEP.TODO_ID, todoRecord.getId())
+                        .execute();
+            }
+
+            assert todoRecord != null;
+            List<Step> savedSteps = context.selectFrom(STEP)
+                    .where(STEP.TODO_ID.eq(todoRecord.getId()))
+                    .fetchInto(Step.class);
+
+            return Todo.builder(todoRecord.getName())
+                    .description(todoRecord.getDescription())
+                    .priority(Priority.valueOf(todoRecord.getPriority().name()))
+                    .steps(savedSteps)
+                    .build();
+        } else throw new ValidationException("The name, or description for the todo " + todo.name()
+                + "with todoId: : {" + todo.id() + "} did not pass the validation");
+    }
+
+    @Transactional
+    @Override
+    public void updateTodo(Long todoId, Todo todo) throws ValidationException {
+        System.out.println("updateTodo() method call");
+        if (context.selectFrom(TODO).where(TODO.ID.eq(todoId)).fetchOne() != null) {
+            var updatedRecord = context.update(TODO)
+                    .set(TODO.NAME, todo.name())
+                    .set(TODO.DESCRIPTION, todo.description())
+                    .set(TODO.PRIORITY, PriorityEnum.valueOf(todo.priority().name()))
+                    .where(TODO.ID.eq(todoId))
+                    .execute();
+
+            if (updatedRecord > 0) {
+                System.out.println("Todo using todoId: {" + todoId + "} has been updated.");
+            } else {
+                throw new ValidationException("Update operation failed for todo with todoId: {" + todoId + "}");
+            }
+        } else {
+            throw new ValidationException("No todo exists under todoId: {" + todoId + "}");
+        }
+    }
+
+    @Transactional
```

Comment:
Almost all of that needs to be refactor. You're mixing validation/business rules that should be in the TodoService, with the data access from/to db. This class should only allow to accces the database, nothing else